components/dot_menu/dot_menu.jsx

```diff
@@ -81,17 +79,8 @@ export default class DotMenu extends Component {
     }
 
     componentDidMount() {
-        $('#' + this.props.idPrefix + '_dropdown' + this.props.post.id).on('shown.bs.dropdown', this.handleDropdownOpened);
-        $('#' + this.props.idPrefix + '_dropdown' + this.props.post.id).on('hidden.bs.dropdown', () => this.props.handleDropdownOpened(false));
-    }
-
-    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
-        if (nextProps.post !== this.props.post) {
-            this.setState({
-                canDelete: PostUtils.canDeletePost(nextProps.post),
-                canEdit: PostUtils.canEditPost(nextProps.post, this.editDisableAction),
-            });
-        }
+        $('#dropdown_' + this.props.post.id).on('shown.bs.dropdown', this.handleDropdownOpened);
+        $('#dropdown_' + this.props.post.id).on('hidden.bs.dropdown', () => this.props.handleDropdownOpened(false));
```

Comment:
Decided against fixing this now. But this does cause a performance issue.

Reply:
What's the performance issue here?

Reply:
All the menus mount and render, they are just hidden by CSS.

Reply:
All jquery calls in dot menu are a major pain in rendering post list to be honest...

Reply:
I don't see an easy solution here but you can at least replace this with $(this.refs.dropdown) and make a ref...should be a bit better